src/lib/parameters/parameters.cpp

```diff
@@ -77,23 +82,39 @@ using namespace time_literals;
 #include "parameters_ioctl.h"
 #endif
 
+
 #if defined(FLASH_BASED_PARAMS)
 #include "flashparams/flashparams.h"
-#else
+#elif !defined(CONFIG_PARAM_CLIENT)
 inline static int flash_param_save(param_filter_func filter) { return -1; }
 inline static int flash_param_load() { return -1; }
 inline static int flash_param_import() { return -1; }
 #endif
 
+// uORB topics needed to keep parameter server and client in sync
+#if defined(CONFIG_PARAM_SERVER)
+#include "param_server.h"
+static char *param_default_file = (char *) "/data/px4/param/parameters";
+static char *param_backup_file = (char *) "/data/px4/param/backup_parameters";
+#elif defined(CONFIG_PARAM_CLIENT)
+#include "param_client.h"
+static char *param_default_file = (char *) "./param/parameters";
```

Comment:
- why does the param client need a file and backup file at all?
 - why should the `param_default_file` and `param_backup_file` be any different between CONFIG_PARAM_SERVER and regular usage?

Reply:
You are correct there is no need for param client to depend on the backup file. Good catch. Has been removed in the most recent push. And currently the directory is statically made in the parameters.cpp file because the PARAM_FILE variable that is used to setup the location, we have yet to setup yet for voxl2.